CRM/Pledge/Page/UserDashboard.php

```diff
@@ -35,25 +35,20 @@ public function listPledges() {
     $controller->process();
     $controller->run();
 
-    // add honor block.
-    $honorParams = [];
+    // Add honor block.
     $honorParams = CRM_Pledge_BAO_Pledge::getHonorContacts($this->_contactId);
-    if (!empty($honorParams)) {
-      // assign vars to templates
-      $this->assign('pledgeHonorRows', $honorParams);
-      $this->assign('pledgeHonor', TRUE);
-    }
-    $session = CRM_Core_Session::singleton();
-    $loggedUserID = $session->get('userID');
-    $this->assign('loggedUserID', $loggedUserID);
+    $this->assign('pledgeHonorRows', $honorParams);
+    $this->assign('pledgeHonor', !empty($honorParams));
+    $this->assign('loggedUserID', CRM_Core_Session::getLoggedInContactID());
   }
 
   /**
-   * the main function that is called when the page
-   * loads, it decides the which action has to be taken for the page.
+   * The main function that is called when the page loads.
+   *
+   * @throws \CRM_Core_Exception
    */
-  public function run() {
-    parent::preProcess();
+  public function run(): void {
+    $this->preProcess();
```

Comment:
As IDE points out this is more correct as it will call the parent anyway, there being no override